eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -37,6 +37,8 @@ stages:
           CtestRegex: ${{ parameters.CtestRegex }}
           Coverage: ${{ parameters.Coverage }}
           CoverageReportPath: ${{ parameters.CoverageReportPath }}
+          ${{ if eq(parameters.ServiceDirectory, 'template') }}:
```

Comment:
We decided to not specify this in the template `ci.yml` because people copied that when starting new pipelines. Doing this here was just a way to prevent others from mistakenly setting up new pipeline as `TestPipline`